File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const sortKeyGroupMap = {};
    for(let str of strs) {
        const sortStr = str.split("").sort().join("");
        if(sortKeyGroupMap[sortStr]) {
            sortKeyGroupMap[sortStr].push(str);
        } else {
            sortKeyGroupMap[sortStr] = [str];
        }
    }
    return Object.values(sortKeyGroupMap);
};

File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> keyAnagrams;
        
        int n = strs.size();
        for(int i=0; i<n; i++) {
            string str = strs[i];
            sort(str.begin(), str.end());
            keyAnagrams[str].push_back(strs[i]);
        }

        vector<vector<string>> ans;
        for(auto& entry: keyAnagrams) {
            ans.push_back(entry.second);
        }
        return ans;
    }
};